Rework README, Quickstart, Comarison (#9)
spxp/spxp-specs: info/Comparison.md

# SPXP compared to other protocols
Multiple other protocols have been proposed for decentralised social networks. It is worth comparing SPXP against a selection
of popular alternatives, investigate the differences and discuss why these do not fulfil the design goals we are looking for
with SPXP:
* individual sovereignty
* privacy
* security
* simplicity

To see how SPXP addresses these requirements, see [here](./SPXP-in-a-nutshell.md). There is a list of challenges and features
NOT supported by SPXP at the bottom of this page.

> ### ActivityPub (gnuscial, Mastodon, Pleroma, Pixelfed, etc)

Reviewer: ```suggestion
### ActivityPub (gnusocial, Mastodon, Pleroma, Pixelfed, etc)
```

[ActivityPub](https://www.w3.org/TR/2018/REC-activitypub-20180123/) uses a federated architecture. Users can freely chose their

> "home instance" from multiple available instance. To participate, the user needs to create an account on this home instance. It

Reviewer: ```suggestion
"home instance" from multiple available instance. To participate, the user needs to create an account on this home instances. It
```

  with you and you won't even see their content. This content moderation has some positive impacts, e.g. for blocking spam and fighting harassment.  
  However, since you don't have control over this content moderation, it is a violation of the *individual sovereignty* we are interested in.
* The authenticity of your data is not guaranteed  
  Without real end-to-end digital signatures, it is possible to tamper with existing messages or completely forge them.  
  This is a violation of the *security* we are interested in.
* Your data is fully accessible by instance owners  
  This includes absolutely everything and is a severe violation of the *privacy* we are interested in.
* You cannot control how your content is used  
  As a federated protocol, it requires your data to be replicated between different instances. This also happens outside of your control, e.g. if
  someone "boosts" (re-tweets) your post. Other instances in the Fediverse will display your content on their website, under their own terms and
  under their exclusive control.  
  This is a violation of the *individual sovereignty* we are interested in.
* You cannot limit the visibility of your content  

>   There is some content you are happy to share with the whole world. But there might be other content you only want to share with your

Reviewer: ```suggestion
  Sure, there is content you are happy to share with the whole world. But there might be other content you only want to share with your
```

* You can probably not control how your content is used  
  At this point, there is only one instance of a PDS. How federation between different PDS is going to work is not yet fully clear. As a federated
  protocol, it is very likely that it will have similar characteristics as ActivityPub.
  This is a violation of the *individual sovereignty* we are interested in.
* You cannot limit the visibility of your content  
  This does not provide the level of *privacy* and *security* we are interested in.

The AT protocol is pretty new and is [currently seeing a lot of changes](https://github.com/bluesky-social/atproto/commits/main). It is
possible that the above analysis is no longer accurate. We will try to keep up with the recent developments and update this section accordingly.

However, as a federated architecture, it suffers from the same generic problem: Real privacy can only be guaranteed by end-to-end encryption,
which prevents the PDS from managing the profile for you.

### Nostr
[Nostr (Notes and Other Stuff Transmitted by Relays)](https://github.com/nostr-protocol/nostr) is an interesting candidate. It solves many

> of our requirements in a quite similar manner. However, it only provides public posts and does not go as far as SPXP in terms for privacy

Reviewer: ```suggestion
of our requirements in a quite similar manner. However, it only provides public posts and does not go as far as SPXP in terms of privacy
```

and sovereignty matters:
* Profiles are bound to a cryptographic keypair owned by the user, making them portable
* Information is published through relays  
  You can freely chose an arbitrary set of relays to publish. Other users will receive your updates as long as they listen to at least
  one of the relays you are publishing on.
* Events are end-to-end signed, guaranteeing message authenticity
* All messages are public, except for direct messages  
  This does not provide the level of *privacy* and *security* we are interested in.
* You cannot limit the visibility of your content  
  This does not provide the level of *privacy* and *security* we are interested in.

> * Relays know too much information about you  

Reviewer: ```suggestion
* Relays have too much information about you  
```

  This does not just includes public messages (obviously), but also metadata of private messages, all profiles you are following and all
  discussion threads you are reading (via filter conditions).  
  This does not provide the level of *privacy* we are interested in.
* No guaranteed event durability  
  Relays can chose to keep events only for a limited period of time
* unproven scalability   
  While this protocol and it's relay architecture is pretty new, it has not yet proven that it can operate at scale. This includes
  horizontal scalability where  clients need to deal with a landscape of thousands of relays as well as vertical scalability where

>   relays need to handle millions of clients consurrently.

Reviewer: ```suggestion
  relays need to handle millions of clients concurrently.
```

One of the main critics with the big established socal networks is the vast collection of user data. There is no difference to Nostr relays.
The same amount of data is generated at relays and owner can mine, analyse, use and sell this information.

## Challenges and requirements for SPXP

By radically pushing power and control from the network into endpoint devices, we face a new hard requirement:  
You need a fat client to fully participate.

It is possible to read and display public content with a very simple client. In order to fully participate with your own profile in
connections, to read private encrypted posts and to contribute with comments to other profiles, the client needs to implement a fair
amount of logic.

With ActivityPub, a simple web browser is laready sufficient. This does not work with SPXP.

## Common features not (yet) provided by SPXP

There are a couple of features commonly provided by social networks which are not available in SPXP:

**Sharing / Re-tweet / Boost**  
It is quite common in popular social networks to re-post an existing message on your own account. With digitally signed posts, SPXP

> could even provide fully authenticated re-posts by including the entire original message.  

Reviewer: ```suggestion
could provide fully authenticated re-posts by including the entire original message.  
```

However, we have not yet made a final decision wether this feature has more good or bad implications. It might be added at a later time.

**Publishing replies on your own feed**  
In other social networks, it is quite common that your followers can see your replies to other posts on your profile. With SPXP, we
made the deliberate choice that the entire discussion is "owned" by the author of the original post. It is possible, and quite
common, that these replies will be encrypted and are only visible to a limited audience.  

> If you follow both profiles, you will be able to see these replies and your client can notify you accordingly.

Reviewer: ```suggestion
If your profile is connected with both, the authoring and the replying profile, then you will be able to see the original post and the reply and your client can notify you accordingly.
```
